### PR TITLE
[Jobs] propagate main api server config to remote jobs controller

### PR DESCRIPTION
**Pull request by rohansonecha**
This PR addresses https://github.com/skypilot-org/skypilot/issues/7957. I was able to minimally reproduce the issue on master with a local api server with the following config:

```
workspaces:
  team-a:
    nebius:
      disabled: true
```
and running `sky jobs launch --config active_workspace=team-a echo hi -y` results in the job being stuck in `PENDING` and the same `ValueError` is seen in the remote controller logs.

The issue is that the `'workspaces'` section of the config on the main api server is not propagated to a remote jobs controller due to `SKIPPED_CLIENT_OVERRIDE_KEYS` in `sky/skylet/constants.py`.

We address this issue by writing a subset (only includes the `'workspaces'` section for now) of the main api server's config to `~/.sky/config.yaml` on the remote jobs controller to configure the local api server that the jobs controller uses to launch managed job clusters.

I verified that the repro case above succeeds with the code changes in this PR. 

I also verified that the above config exists on the remote jobs controller at `~/.sky/config.yaml` and making changes to the `'workspaces'` section of the config on the local api server and then running the same `sky jobs launch` command as above also updates the `~/.sky/config.yaml` file on the remote controller.

Finally, I made sure that changes to the config that are outside of the `'workspaces'` section of the config are not propagated to the `~/.sky/config.yaml` file on the remote controller. 

Tested (run the relevant ones):

- [x] Code formatting: install pre-commit (auto-check on commit) or `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `/smoke-test` (CI) or `pytest tests/test_smoke.py` (local)
- [ ] Relevant individual tests: `/smoke-test -k test_name` (CI) or `pytest tests/test_smoke.py::test_name` (local)
- [ ] Backward compatibility: `/quicktest-core` (CI) or `pytest tests/smoke_tests/test_backward_compat.py` (local)

### CODE REVIEW

File: sky/templates/jobs-controller.yaml.j2 [hunk 1]
```diff
@@ -14,6 +14,9 @@ file_mounts:
   {%- for controller_file_mount_path, local_file_mount_path in local_to_controller_file_mounts.items() %}
   {{controller_file_mount_path}}: {{local_file_mount_path}}
   {%- endfor %}
+  {%- if consolidation_mode_job_id is none %}
+  {{ '~/.sky/config.yaml' }}: {{ '~/.sky/config.yaml' }}
```
Comment (Michaelvll) [inline sky/templates/jobs-controller.yaml.j2]: Hmm, the source doesn't make sense, as the user could use a custom env var for the config yaml right?

Comment (Michaelvll) [reply sky/templates/jobs-controller.yaml.j2]: Also, there could be project level config, etc. We should not directly use the path

Comment (Michaelvll) [reply sky/templates/jobs-controller.yaml.j2]: We need to double check that if it is ok to upload the entire config YAML. It could affect the robustness of managed jobs if there are some configs that should not be uploaded as a global config.

Comment (rohansonecha) [reply sky/templates/jobs-controller.yaml.j2]: This was meant to be a draft PR just to get eyes from @cg505 on the direction I am going with this (updated the status to Draft now for clarity). To address your concerns:

1. Yep, I plan to use `get_server_config()` from `sky/skypilot_config.py` rather than hard-coding the file path as the source.
2. I don't think the project level config is relevant here as it is a client-side concept, and we only want to pass the main api server's config to the remote jobs controller's local api server.
3. I plan to add some logic to only propagate a subsection of the config, starting with only sending the `'workspaces'` section. We can send over more subsections of the config as needed.

File: sky/templates/jobs-controller.yaml.j2 [hunk 2]
```diff
@@ -14,6 +14,9 @@ file_mounts:
   {%- for controller_file_mount_path, local_file_mount_path in local_to_controller_file_mounts.items() %}
   {{controller_file_mount_path}}: {{local_file_mount_path}}
   {%- endfor %}
+  {%- if consolidation_mode_job_id is none %}
+  {{ '~/.sky/config.yaml' }}: {{ mutated_local_api_server_config_path }}
```
Comment (cg505) [inline sky/templates/jobs-controller.yaml.j2]:
```suggestion
  ~/.sky/config.yaml: {{ mutated_local_api_server_config_path }}
```

**Comment by Michaelvll** [comment]
Please test this well with manual edge cases.

File: sky/templates/jobs-controller.yaml.j2 [hunk 3]
```diff
@@ -14,6 +14,9 @@ file_mounts:
...
+  {%- endif %}
```
Comment (aylei) [inline sky/templates/jobs-controller.yaml.j2:19]: Changing the global config in job controller has a large impact radius and seems concerning to me. Curious, why do we need to set global config in this case instead of setting workspace to the config file dedicated to that managed job: https://github.com/skypilot-org/skypilot/blob/fb4ddc07a173a32309b772fd05e7e88adb4c107c/sky/utils/controller_utils.py#L553

Comment (aylei) [reply sky/templates/jobs-controller.yaml.j2:19]: discussed offline, addressed

**Comment by rohansonecha** [comment]
closing for https://github.com/skypilot-org/skypilot/pull/7966
